src/main/java/bio/terra/service/resourcemanagement/azure/AzureMonitoringService.java

```diff
@@ -0,0 +1,626 @@
+                by AccountName, CallerIpAddress, Identity, bin(TimeGenerated, 10m)
+            | where Attempts > 10
+            | project TimeStart, TimeEnd, Attempts, CallerIpAddress, AccountName, Identity
+                                        """)
+                .withSuppressionEnabled(false)
+                .withSuppressionDuration(Duration.parse("PT1H"))
+                .withQueryPeriod(Duration.ofDays(1))
+                .withQueryFrequency(Duration.ofDays(1))
+                .withEnabled(true)
+                .withSeverity(AlertSeverity.INFORMATIONAL)
+                .withTriggerOperator(TriggerOperator.GREATER_THAN)
+                .withTriggerThreshold(0))
+        .id();
+  }
+
+  /**
+   * Delete the Sentinel alert rule for the Sentinel instance monitoring the given Storage Account.
+   * Note: this is for the rule that track unauthorized access to the storage account's files
+   *
+   * @param profileModel The billing profile for the dataset or snapshot being logged
+   * @param storageAccount The Storage Account being monitored
+   */
+  public void deleteSentinelRuleUnauthorizedAccess(
+      BillingProfileModel profileModel, AzureStorageAccountResource storageAccount) {
+    SecurityInsightsManager client =
+        resourceConfiguration.getSecurityInsightsManagerClient(
+            resourceConfiguration.getCredentials().getHomeTenantId(),
+            profileModel.getSubscriptionId());
+
+    logger.info(
+        "Deleting Sentinel UnauthorizedAccess alert rule for Storage Account {}",
+        storageAccount.getStorageAccountId());
+    client
+        .alertRules()
+        .delete(
+            storageAccount.getApplicationResource().getAzureResourceGroupName(),
+            storageAccount.getName(),
+            UNAUTHORIZED_ACCESS_ALERT_NAME);
+  }
+
+  /**
+   * Retrieve the rule for sending Slack notifications
+   *
+   * @param profileModel The billing profile for the dataset or snapshot being monitored
+   * @param storageAccount The Storage Account being monitored
+   * @return A {@link AutomationRule} object representing the Sentinel notification rule or null if
+   *     it doesn't exist
+   */
+  public AutomationRule getNotificationRule(
+      BillingProfileModel profileModel, AzureStorageAccountResource storageAccount) {
+    SecurityInsightsManager client =
+        resourceConfiguration.getSecurityInsightsManagerClient(
+            resourceConfiguration.getCredentials().getHomeTenantId(),
+            profileModel.getSubscriptionId());
+    try {
+      AutomationRule automationRule =
+          client
+              .automationRules()
+              .get(
+                  storageAccount.getApplicationResource().getAzureResourceGroupName(),
+                  storageAccount.getName(),
+                  SLACK_ALERT_RULE_NAME);
+      logger.debug("Found Sentinel alert rule");
+      return automationRule;
+    } catch (ManagementException e) {
```

Comment:
This appears to be a common pattern for calling azure methods that can throw `ManagementException`. Could this be abstracted into a method? It looks like the general rule is
- log message with exception
- compare against a list of code values (could use set contains to represent this concept rather than `equals() || equals()`
- in some cases, `&&` with a `e.getMessage().contains()`

Reply:
For example:

```
  private static <T> T callApi(Supplier<T> apiCall, String message, Set<String> codes) {
    try {
      return apiCall.get();
    } catch (ManagementException e) {
      logger.debug(message, e);
      if (codes.contains(e.getValue().getCode())) {
        return null;
      } else {
        throw e;
      }
    }
  }

...

  public AutomationRule getNotificationRule(
      BillingProfileModel profileModel, AzureStorageAccountResource storageAccount) {
    SecurityInsightsManager client =
        resourceConfiguration.getSecurityInsightsManagerClient(
            resourceConfiguration.getCredentials().getHomeTenantId(),
            profileModel.getSubscriptionId());

    return callApi(() -> {
      AutomationRule automationRule =
          client
              .automationRules()
              .get(
                  storageAccount.applicationResource().getAzureResourceGroupName(),
                  storageAccount.name(),
                  SLACK_ALERT_RULE_NAME);
      logger.debug("Found Sentinel alert rule");
      return automationRule;
    }, "No Sentinel alert rule", Set.of(RESOURCE_NOT_FOUND_CODE, NOT_FOUND_CODE));
  }
```

Reply:
Yeah, I was kinda torn between abstracting that out vs. keeping the individual methods individually readable (also, if you noticed, some of the handling is different based on the base client...but I think just doing the most complex one is pretty cheap).  All that to say, I agree with you.  I'll make the change

Reply:
I'll add to a new class (`AzureResourceUtils`)

Reply:
Ugh, you know what, they're different enough that I'm not sure it's worth pulling into a common method

Reply:
Other than the case that handles `getStatusCode()`, they all seem equivalent to me. They all log on the positive and negative cases, so that's another thing that could be refactored. E.g.

```
  private static <T> T callApi(
      Supplier<T> apiCall, String foundMessage, String notFoundMessage, Set<String> codes) {
    return callApi(apiCall, foundMessage, notFoundMessage, codes, null);
  }

  private static <T> T callApi(
      Supplier<T> apiCall,
      String foundMessage,
      String notFoundMessage,
      Set<String> codes,
      String matchMessage) {
    try {
      T result = apiCall.get();
      logger.info(foundMessage);
      return result;
    } catch (ManagementException e) {
      logger.warn(notFoundMessage, e);
      if (codes.contains(e.getValue().getCode())
          && (matchMessage == null || e.getMessage().contains(matchMessage))) {
        return null;
      } else {
        throw e;
      }
    }
  }

  public AutomationRule getNotificationRule(
      BillingProfileModel profileModel, AzureStorageAccountResource storageAccount) {
    SecurityInsightsManager client =
        resourceConfiguration.getSecurityInsightsManagerClient(
            resourceConfiguration.getCredentials().getHomeTenantId(),
            profileModel.getSubscriptionId());
    return callApi(
        () ->
            client
                .automationRules()
                .get(
                    storageAccount.getApplicationResource().getAzureResourceGroupName(),
                    storageAccount.getName(),
                    SLACK_ALERT_RULE_NAME),
        "Found Sentinel alert rule",
        "No Sentinel alert rule",
        Set.of(RESOURCE_NOT_FOUND_CODE, NOT_FOUND_CODE));
  }
```
I see five common cases of this form in this file.

Reply:
There's some subtle differences. Some do a check with the contents of the error message.  Some check the return code. I find that it makes it more confusing if it's not consistent

Reply:
There's only one that checks the return code, but there are five others that share common code.

Reply:
Not quite.  `getSentinelRuleUnauthorizedAccess` also checks for the content of the message (which I know is gross but that's what we get from the API :( )

Reply:
Yes, the code snippet I posted above handles that case as well.